zerver/tests/test_retention.py

```diff
@@ -21,95 +37,452 @@ class TestRetentionLib(ZulipTestCase):
+        exp_messages_ids.extend([msg.id for msg in exp_mit_msgs.order_by('id')])
+        self._make_mit_msgs(4, timezone.now() - timedelta(days=50))
+        zulip_msgs_ids = list(Message.objects.order_by('id').filter(
+            sender__realm=self.zulip_realm).values_list('id', flat=True)[3:10])
+        exp_messages_ids.extend(zulip_msgs_ids)
+        exp_zulip_msgs = self._change_msgs_pub_date(zulip_msgs_ids,
+                                                    timezone.now() - timedelta(days=31))
+        move_expired_messages_to_archive()
+        move_expired_user_messages_to_archive()
+        self.assertEqual(ArchiveMessage.objects.count(), len(exp_messages_ids))
+        self.assertEqual(
+            ArchiveUserMessage.objects.count(),
+            UserMessage.objects.filter(message_id__in=exp_messages_ids).count()
         )
+        # Compare expected messages ids with archived messages by realm.
+        self._check_archive_data_by_realm(exp_mit_msgs, self.mit_realm)
+        self._check_archive_data_by_realm(exp_zulip_msgs, self.zulip_realm)
 
     def test_expired_messages_in_one_realm(self):
         # type: () -> None
         # Check realm with expired messages and messages
         # with one day to expiration data.
-        expired_mit_messages = self._make_mit_messages(5, timezone.now() - timedelta(days=101))
-        actual_mit_messages = self._make_mit_messages(3, timezone.now() - timedelta(days=99))
-        expired_messages_result = list(get_expired_messages())
-        expired_mit_messages_ids = [message.id for message in expired_mit_messages]
-        expired_mit_messages_result_ids = [message.id for message in
-                                           expired_messages_result[0]['expired_messages']]
-        actual_mit_messages_ids = [message.id for message in actual_mit_messages]
-        self.assertEqual(len(expired_messages_result), 1)
-        self.assertEqual(len(expired_messages_result[0]['expired_messages']), 5)
-        self.assertEqual(expired_messages_result[0]['realm_id'], self.mit_realm.id)
-        # Compare expected messages ids with result messages ids.
-        self.assertEqual(
-            sorted(expired_mit_messages_ids),
-            expired_mit_messages_result_ids
-        )
-        # Check actual mit.edu messages are not contained in expired messages list
-        self.assertEqual(
-            set(actual_mit_messages_ids) - set(expired_mit_messages_ids),
-            set(actual_mit_messages_ids)
+        exp_mit_msgs = self._make_mit_msgs(5, timezone.now() - timedelta(days=101))
+        move_expired_messages_to_archive()
+        move_expired_user_messages_to_archive()
+        archived_user_messages = ArchiveUserMessage.objects.all()
+        self.assertEqual(ArchiveMessage.objects.count(), 5)
+        self.assertEqual(archived_user_messages.count(), 10)
+        # Compare expected messages ids with archived messages in mit realm
```

Comment:
I think this should be:

`# Compare expected messages' IDs with IDs of archived messages in MIT realm`